### DISCUSSION TITLE
Restrict email change, or limit to one unused.

### DISCUSSION THREAD

**Discussion by tetricky** [lldap/lldap#466] [Q&A] [answered]
I have a service where the authenticated user is defined by the email address (ie the email address is authoritative, not the username).

A consequence of this is that a user can register with the service, change the email address within lldap to that of an existing user (for example an administrative user), and then perform actions based on a different existing user with that pre-existing email address (defined by the user with the changed email address).

There does not seem to be a way to prevent that change, so I can restrict access to lldap, which adds a significant administration overhead and eliminates the possibility for users to modify their password and display name. Which is sub-optimal. Or rely on compliant behaviours. Which is a complete nightmare scenario.

The best solution for me would be to restrict email reset to only email addresses not yet used (and/or validate them before changing).

The next best solution would be to be able to set the email address only with `lldap_admin` privilege, not with `lldap_password_manager`.

Is anything like this possible with the current status of the project? If not is there a route to achieve this, or something equivalent?

**Comment by nitnelave** [comment, marked as answer]
I think it's fair to require that the email addresses are unique. That's
the easiest thing I can think of. Would that be enough for your use case?

**Reply by nitnelave** [reply]
See https://github.com/lldap/lldap/pull/552

**Comment by tetricky** [comment]
Yes. That would stop people commandeering an existing username (by spoofing the email address through lldap). I think it would be the tidiest (simple, in terms of user interaction, most elegant) solution.
